Migration to PHP 8

bastien70 opened the pull request:
Some modifications to port the code to PHP 8

1. Asserts annotations to PHP attributes
2. replaced some php functions by php 8 equivalent
3. Added properties type

Sorry I haven't had time to do other things yet. I will try as soon as possible to add commits against the return types

bastien70 committed:
Migrated assert annotations to PHP attributes

bastien70 committed:
Replaced strpos() by str_contains() for PHP 8 migration

bastien70 committed:
Changed switch to match for PHP 8 migration

bastien70 committed:
Added properties type

jmsche commented:
Hi, thanks for the PR!

Can you please fix the runs? I guess removing PHP 7.4 from matrix & switching other workflows to PHP 8.0 should do the trick.

bastien70 committed:
Updated workflows

jmsche commented on composer.json:
```diff
@@ -46,7 +46,8 @@
         "phpstan/phpstan": "^0.12.59",
-        "symfony/phpunit-bridge": "^4.4 || ^5.0 || ^6.0"
+        "symfony/phpunit-bridge": "^4.4 || ^5.0 || ^6.0",
+        "rector/rector": "^0.11.60"
```
Please, don't add Rector :)

jmsche commented on rector.php:
```diff
@@ -0,0 +1,32 @@
+<?php
```
Please, don't add Rector :)

jmsche commented on src/Controller/FeedController.php:
```diff
@@ -13,20 +13,11 @@
  */
```
Class comment can be removed

bastien70 committed:
Removed Rector

jmsche commented on src/Controller/FeedController.php:
```diff
@@ -13,20 +13,11 @@
-    private $twig;
+    private Environment $twig;
 
     public function __construct(FeedManager $feedManager, ValidatorInterface $validator, Environment $twig)
```
Please use constructor property promotion

jmsche commented on src/Controller/SitemapController.php:
```diff
@@ -14,25 +14,10 @@
+    private Environment $twig;
+    private HttpKernelInterface $httpKernel;
 
     public function __construct(SitemapManager $sitemapManager, RouterInterface $router, Environment $twig, HttpKernelInterface $httpKernel)
```
Constructor property promotion

jmsche commented on src/DQL/BitwiseAnd.php:
```diff
@@ -12,15 +12,9 @@
-     * @var \Doctrine\ORM\Query\AST\Literal
-     */
-    private $firstValue;
+    private \Doctrine\ORM\Query\AST\Literal $firstValue;
```
Import the class?

jmsche commented on src/Datalist/Action/DatalistAction.php:
```diff
@@ -9,15 +9,9 @@
-    private $datalist;
+    private DatalistInterface $datalist;
 
     public function __construct(DatalistActionConfig $config)
```
Constructor property promotion

jmsche commented on src/Datalist/Action/Type/SimpleActionType.php:
```diff
@@ -13,10 +13,7 @@
-     * @var \Symfony\Component\Routing\RouterInterface
-     */
-    protected $router;
+    protected RouterInterface $router;
```
Constructor property promotion

jmsche commented on src/Datalist/Config.php:
```diff
@@ -70,10 +59,9 @@ public function getOption($name, $default = null)
      * @param $value
      */
-    public function setOption($name, $value)
+    public function setOption(string $name, $value)
```
mixed?

jmsche commented on src/Datalist/Datalist.php:
```diff
@@ -16,85 +16,37 @@
-    private $routeParams = [];
+    private array $routeParams = [];
 
     public function __construct(DatalistConfig $config)
```
Constructor property promotion

jmsche commented on src/Datalist/DatalistBuilder.php:
```diff
@@ -19,30 +19,15 @@
-     * @var FormFactoryInterface
-     */
-    private $formFactory;
+    private FormFactoryInterface $formFactory;
```
Constructor property promotion

jmsche commented on src/Datalist/Datasource/ArrayDatasource.php:
```diff
@@ -162,44 +156,20 @@ private function buildComparisonExpressionCallback(ComparisonExpression $express
-                    throw new \UnexpectedValueException(sprintf('Unknown operator "%s"', $operator));
-                    break;
-            }
+            $result = match ($operator) {
```
Direct return

jmsche commented on src/Datalist/Datasource/DoctrineORMDatasource.php:
```diff
@@ -13,15 +13,9 @@
-    private $initialized = false;
+    private bool $initialized = false;
 
     public function __construct(QueryBuilder $queryBuilder)
```
Constructor property promotion

jmsche commented on src/Datalist/Field/DatalistField.php:
```diff
@@ -12,15 +12,9 @@
-    private $datalist;
+    private DatalistInterface $datalist;
 
     public function __construct(DatalistFieldConfig $config)
```
Constructor property promotion

jmsche commented on src/Datalist/Filter/DatalistFilter.php:
```diff
@@ -6,15 +6,12 @@
-    private $datalist;
+    private DatalistInterface $datalist;
 
     public function __construct(DatalistFilterConfig $config)
```
Constructor property promotion

jmsche commented on src/Datalist/Filter/DatalistFilterExpressionBuilder.php:
```diff
@@ -10,7 +10,7 @@ class DatalistFilterExpressionBuilder
      * @var Expression\CombinedExpression
      */
-    private $expression;
+    private CombinedExpression $expression;
```
Docblock can be removed

jmsche commented on src/Datalist/Filter/Expression/CombinedExpression.php:
```diff
@@ -7,20 +7,11 @@ class CombinedExpression implements ExpressionInterface
-     * @param string $operator
-     */
-    public function __construct($operator)
+    public function __construct(string $operator)
```
Constructor property promotion

jmsche commented on src/Datalist/Filter/Expression/ComparisonExpression.php:
```diff
@@ -16,29 +16,16 @@ class ComparisonExpression implements ExpressionInterface
      * @throws \InvalidArgumentException
      */
-    public function __construct($propertyPath, $operator, $value)
+    public function __construct(string $propertyPath, string $operator, mixed $value)
```
Constructor property promotion

jmsche commented on src/Doctrine/Mapping/File.php:
```diff
@@ -11,15 +11,19 @@
-    public $nameCallback;
+    public ?string $nameCallback = null;
 
     public function __construct(string $path = null, string $pathCallback = null, string $mappedBy = null, string $filename = null, string $nameCallback = null)
```
Constructor property promotion

jmsche commented on src/Feed/FeedItem.php:
```diff
@@ -13,50 +13,44 @@ class FeedItem
      *
-     * @Assert\NotBlank()
-     *
      * @var string
```
Some docblocks can be removed (eg. `@var string`)

jmsche commented on src/Feed/FeedManager.php:
```diff
@@ -7,31 +7,24 @@
+    protected array $feeds = [];
 
     /**
      * Register a feed in the manager.
```
Can be removed I guess

bastien70 replied:
What do you want to remove ?

jmsche replied:
The comment "Register a feed in the manager."

jmsche commented on src/Form/Type/AbstractRecaptchaType.php:
```diff
@@ -12,38 +12,30 @@ abstract class AbstractRecaptchaType extends AbstractType
      * @param string $apiHost   Api host
      */
-    public function __construct($publicKey, $enabled, $apiHost = 'www.google.com')
+    public function __construct(string $publicKey, bool $enabled, string $apiHost = 'www.google.com')
```
Constructor property promotion

jmsche commented on src/Form/Type/RecaptchaType.php:
```diff
@@ -12,13 +12,10 @@ class RecaptchaType extends AbstractRecaptchaType
-    protected $localeResolver;
+    protected LocaleResolver $localeResolver;
 
     public function __construct($publicKey, $enabled, $ajax, LocaleResolver $localeResolver, $apiHost = 'www.google.com')
```
Constructor property promotion

jmsche commented on src/Listener/FileSubscriber.php:
```diff
@@ -21,30 +21,15 @@
-     * @param string $uploadDir
-     */
-    public function __construct($uploadDir)
+    public function __construct(string $uploadDir)
```
Constructor property promotion

jmsche commented on src/Locale/LocaleResolver.php:
```diff
@@ -9,26 +9,13 @@
-     * @param bool   $useLocaleFromRequest
-     */
-    public function __construct($defaultLocale, $useLocaleFromRequest, RequestStack $requestStack)
+    public function __construct(string $defaultLocale, bool $useLocaleFromRequest, RequestStack $requestStack)
```
Constructor property promotion

jmsche commented on src/Navigation/NavigationRegistry.php:
```diff
@@ -9,20 +9,11 @@
-    private $breadcrumbsPaths = [];
+    private array $breadcrumbsPaths = [];
 
     public function __construct(RequestStack $requestStack)
```
Constructor property promotion

jmsche commented on src/Paginator/ArrayPaginator.php:
```diff
@@ -7,10 +7,7 @@
-     * @var array
-     */
-    private $items;
+    private array $items;
```
Constructor property promotion

jmsche commented on src/Twig/Extension/DatalistExtension.php:
```diff
@@ -16,20 +16,11 @@
-    private $themes;
+    private \SplObjectStorage $themes;
 
     public function __construct(RequestStack $requestStack)
```
Constructor property promotion

jmsche commented on src/Twig/Extension/DateExtension.php:
```diff
@@ -8,10 +8,7 @@
-    private $translator;
+    private TranslatorInterface $translator;
 
     public function __construct(TranslatorInterface $translator)
```
Constructor property promotion

jmsche commented on src/Twig/Extension/FacebookExtension.php:
```diff
@@ -11,10 +11,7 @@
-     * @var string
-     */
-    private $appId;
+    private string $appId;
```
Constructor property promotion

jmsche commented on src/Twig/Extension/GoogleExtension.php:
```diff
@@ -13,30 +13,15 @@ class GoogleExtension extends AbstractExtension
-    private $tagsManagerId;
+    private string $tagsManagerId;
 
     public function __construct($accountId, $debug = false)
```
Constructor property promotion

jmsche commented on src/Twig/Extension/NavigationExtension.php:
```diff
@@ -11,10 +11,7 @@
-    private $registry;
+    private NavigationRegistry $registry;
 
     public function __construct(NavigationRegistry $registry)
```
Constructor property promotion

jmsche commented on src/Twig/Extension/PaginatorExtension.php:
```diff
@@ -15,25 +15,13 @@
-     * @param string $template
-     */
-    public function __construct($template, RequestStack $requestStack)
+    public function __construct(string $template, RequestStack $requestStack)
```
Constructor property promotion

jmsche commented on src/Util/StringUtil.php:
```diff
@@ -93,13 +90,11 @@ public static function slugify($string)
      * @deprecated
      */
-    public static function sluggify($string)
+    public static function sluggify(string $string)
```
You can remove this method

jmsche commented on src/Validator/Constraints/RecaptchaV3Validator.php:
```diff
@@ -13,17 +13,13 @@ class RecaptchaV3Validator extends ConstraintValidator
 
-    /** @var LoggerInterface */
-    private $logger;
+    private LoggerInterface $logger;
```
Constructor property promotion

jmsche commented on src/Validator/Constraints/RecaptchaValidator.php:
```diff
@@ -15,45 +15,30 @@ class RecaptchaValidator extends ConstraintValidator
-     * @param bool   $verifyHost
-     */
-    public function __construct($enabled, $privateKey, RequestStack $requestStack, array $httpProxy, $verifyHost)
+    public function __construct(bool $enabled, string $privateKey, RequestStack $requestStack, array $httpProxy, bool $verifyHost)
```
Constructor property promotion

jmsche reviewed (changes requested):
* Yes, please add return types everywhere :)
* Useless class comments (like "Class BooleanFieldType." can be removed)
* Constructor property promotion can be used in many files

bastien70 committed:
Added constructor property promotions, removed useless class comment

bastien70 commented:
> * Yes, please add return types everywhere :)
> * Useless class comments (like "Class BooleanFieldType." can be removed)
> * Constructor property promotion can be used in many files

1. Okay, I'll take care of it ASAP, in the week I'll try
2. Done
3. Done

bastien70 committed:
Removed FeedManager comment

bastien70 committed:
Added return types

bastien70 commented:
I added the return types.
However, I have not yet removed the annotations from `@return,` in order to make it easier for you for the review to compare the return of the annotation and the one I added myself.

I had to modify several return types here and there due to the errors it was causing.

But it is very possible that unfortunately I missed some return types which were still incorrect.

You can already review this, and then when I spend some time on it, I'll go through each method one by one and check again if the return type looks right.

After that, I will be able to remove the annotations

jmsche commented on src/Controller/FeedController.php:
```diff
@@ -8,41 +8,21 @@
      * @throws \ErrorException
      */
-    public function indexAction(Request $request, $feedName)
+    public function indexAction(Request $request, string $feedName)
```
Missing return type

jmsche commented on src/DQL/BitwiseAnd.php:
```diff
@@ -3,24 +3,16 @@
-    private $secondValue;
+    private Literal $secondValue;
 
     public function parse(Parser $parser)
```
Missing return type

jmsche commented on src/Datalist/Action/DatalistAction.php:
```diff
@@ -62,23 +43,17 @@ public function getOption($name, $default = null)
      * @return mixed
      */
-    public function setDatalist(DatalistInterface $datalist)
+    public function setDatalist(DatalistInterface $datalist): void
```
Would be interesting to return self instead

jmsche commented on src/Datalist/Action/DatalistActionInterface.php:
```diff
@@ -3,33 +3,23 @@
 /**
  * Interface DatalistActionInterface.
  */
 interface DatalistActionInterface
```
Docblock can be removed

jmsche commented on src/Datalist/Action/Type/SimpleActionType.php:
```diff
@@ -8,19 +8,10 @@
-        $this->router = $router;
     }
 
     public function configureOptions(OptionsResolver $resolver)
```
Missing return type

jmsche commented on src/Datalist/Config.php:
```diff
@@ -69,19 +37,15 @@ public function getOption($name, $default = null)
-     * @param $value
-     */
-    public function setOption($name, $value)
+    public function setOption(string $name, mixed $value)
```
Missing return type, maybe return self?

jmsche commented on src/Datalist/Datalist.php:
```diff
@@ -233,101 +161,77 @@ public function getPaginator()
+     * @return FormInterface
      */
-    public function getSearchForm()
+    public function getSearchForm(): FormInterface|Form
```
FormInterface is enough

jmsche commented on src/Datalist/Datalist.php:
```diff
@@ -233,101 +161,77 @@ public function getPaginator()
+     * @return FormInterface
      */
-    public function getFilterForm()
+    public function getFilterForm(): FormInterface|Form
```
FormInterface is enough

jmsche commented on src/Datalist/Datalist.php:
```diff
@@ -370,61 +272,45 @@ public function bind($data)
-     * @return DatalistInterface
      */
-    public function setRoute($route)
+    public function setRoute($route): DatalistInterface
```
Missing param type

jmsche commented on src/Datalist/DatalistFactory.php:
```diff
@@ -51,31 +48,25 @@ public function __construct(FormFactoryInterface $formFactory, RouterInterface $
-     * @return DatalistInterface
      */
-    public function create($type = 'datalist', array $options = [])
+    public function create($type = 'datalist', array $options = []): DatalistInterface
```
Missing param type

jmsche commented on src/Datalist/DatalistFactory.php:
```diff
@@ -51,31 +48,25 @@ public function __construct(FormFactoryInterface $formFactory, RouterInterface $
-     * @return Datalist
      */
-    public function createNamed($name, $type = 'datalist', array $options = [])
+    public function createNamed($name, $type = 'datalist', array $options = []): Datalist
```
Missing param types

jmsche commented on src/Datalist/DatalistFactory.php:
```diff
@@ -88,11 +79,9 @@ public function createBuilder($type = 'datalist', array $options = [])
      * @throws \InvalidArgumentException
      */
-    public function createNamedBuilder($name, $type = 'datalist', array $options = [])
+    public function createNamedBuilder($name, $type = 'datalist', array $options = []): DatalistBuilder
```
Missing param types

jmsche commented on src/Datalist/DatalistFactory.php:
```diff
@@ -116,7 +105,7 @@ public function createNamedBuilder($name, $type = 'datalist', array $options = [
      * @return DatalistType\DatalistTypeInterface
      */
-    public function getType($alias)
+    public function getType($alias): Type\DatalistTypeInterface
```
Missing param type

jmsche commented on src/Datalist/DatalistFactory.php:
```diff
@@ -136,11 +125,9 @@ public function registerType(DatalistType\DatalistTypeInterface $type)
      * @throws \InvalidArgumentException
      */
-    public function getFieldType($alias)
+    public function getFieldType($alias): FieldType\FieldTypeInterface
```
Missing param type

jmsche commented on src/Datalist/DatalistFactory.php:
```diff
@@ -157,11 +144,9 @@ public function registerFieldType(FieldType\FieldTypeInterface $fieldType)
      * @throws \InvalidArgumentException
      */
-    public function getFilterType($alias)
+    public function getFilterType($alias): FilterType\FilterTypeInterface
```
Missing param type

jmsche commented on src/Datalist/DatalistFactory.php:
```diff
@@ -178,11 +163,9 @@ public function registerFilterType(FilterType\FilterTypeInterface $filterType)
      * @throws \InvalidArgumentException
      */
-    public function getActionType($alias)
+    public function getActionType($alias): ActionType\ActionTypeInterface
```
Missing param type

jmsche commented on src/Datalist/DatalistInterface.php:
```diff
@@ -6,40 +6,25 @@
 
 /**
  * Interface DatalistInterface.
  */
```
Docblock can be removed

jmsche commented on src/Datalist/Datasource/AbstractDatasource.php:
```diff
@@ -4,104 +4,61 @@
-     * @var \Leapt\CoreBundle\Paginator\PaginatorInterface
-     */
-    protected $paginator;
+    protected \Leapt\CoreBundle\Paginator\PaginatorInterface $paginator;
```
Import the FQCN instead

jmsche commented on src/Datalist/Datasource/ArrayDatasource.php:
```diff
@@ -93,7 +76,7 @@ protected function initialize()
      * @throws \InvalidArgumentException
      */
-    private function buildExpressionCallback(ExpressionInterface $expression)
+    private function buildExpressionCallback(ExpressionInterface $expression): callable|\Closure
```
I guess callable should be enough

jmsche commented on src/Datalist/Datasource/ArrayDatasource.php:
```diff
@@ -112,7 +95,7 @@ private function buildExpressionCallback(ExpressionInterface $expression)
      * @throws \UnexpectedValueException
      */
-    private function buildCombinedExpressionCallback(CombinedExpression $expression)
+    private function buildCombinedExpressionCallback(CombinedExpression $expression): callable|\Closure
```
I guess callable should be enough

jmsche commented on src/Datalist/Datasource/ArrayDatasource.php:
```diff
@@ -154,54 +137,28 @@ private function buildCombinedExpressionCallback(CombinedExpression $expression)
      * @throws \UnexpectedValueException
      */
-    private function buildComparisonExpressionCallback(ComparisonExpression $expression)
+    private function buildComparisonExpressionCallback(ComparisonExpression $expression): callable|\Closure
```
I guess callable should be enough

jmsche commented on src/Datalist/Datasource/DatasourceInterface.php:
```diff
@@ -9,39 +9,18 @@
-     * @return \Leapt\CoreBundle\Paginator\PaginatorInterface
-     */
-    public function getPaginator();
+    public function getPaginator(): \Leapt\CoreBundle\Paginator\PaginatorInterface;
```
Import FQCN

jmsche commented on src/Datalist/Datasource/DoctrineORMDatasource.php:
```diff
@@ -90,36 +80,34 @@ protected function initialize()
+     * @throws InvalidArgumentException
      */
-    private function buildQueryBuilderExpression(ExpressionInterface $expression)
+    private function buildQueryBuilderExpression(ExpressionInterface $expression): Orx|Andx|Comparison
```
Comparison should be enough

jmsche commented on src/Datalist/Datasource/DoctrineORMDatasource.php:
```diff
@@ -90,36 +80,34 @@ protected function initialize()
-     * @throws \UnexpectedValueException
+     * @throws UnexpectedValueException
      */
     private function buildQueryBuilderCombinedExpression(CombinedExpression $expression)
```
Missing return type

jmsche commented on src/Datalist/Field/DatalistField.php:
```diff
@@ -3,86 +3,51 @@
      * @return mixed|void
      */
-    public function setOption($name, $value)
+    public function setOption(string $name, mixed $value)
```
Missing return type; also return self?

jmsche commented on src/Datalist/Field/DatalistFieldInterface.php:
```diff
@@ -12,48 +13,21 @@ interface DatalistFieldInterface
-     * @param mixed  $default
-     */
-    public function getOption($name, $default = null);
+    public function getOption(string $name, mixed $default = null);
```
Missing return type

jmsche commented on src/Datalist/Field/DatalistFieldInterface.php:
```diff
@@ -12,48 +13,21 @@ interface DatalistFieldInterface
-     * @param mixed  $value
-     */
-    public function setOption($name, $value);
+    public function setOption(string $name, mixed $value);
```
Missing return type; also return self?

jmsche commented on src/Datalist/Field/Type/AbstractFieldType.php:
```diff
@@ -24,10 +21,7 @@ public function configureOptions(OptionsResolver $resolver)
-     * @param mixed $row
-     */
-    public function buildViewContext(ViewContext $viewContext, DatalistFieldInterface $field, $row, array $options)
+    public function buildViewContext(ViewContext $viewContext, DatalistFieldInterface $field, mixed $row, array $options)
```
Missing return type

jmsche commented on src/Datalist/Field/Type/BooleanFieldType.php:
```diff
@@ -25,29 +22,20 @@ public function configureOptions(OptionsResolver $resolver)
-     * @param mixed $row
-     */
-    public function buildViewContext(ViewContext $viewContext, DatalistFieldInterface $field, $row, array $options)
+    public function buildViewContext(ViewContext $viewContext, DatalistFieldInterface $field, mixed $value, array $options)
```
Missing return type

jmsche commented on src/Datalist/Field/Type/DateTimeFieldType.php:
```diff
@@ -4,9 +4,6 @@
- */
 class DateTimeFieldType extends AbstractFieldType
 {
     public function configureOptions(OptionsResolver $resolver)
```
Missing return type

jmsche commented on src/Datalist/Field/Type/LabelFieldType.php:
```diff
@@ -6,9 +6,6 @@
- */
 class LabelFieldType extends AbstractFieldType
 {
     public function configureOptions(OptionsResolver $resolver)
```
Missing return type

jmsche commented on src/Datalist/Field/Type/LabelFieldType.php:
```diff
@@ -22,14 +19,12 @@ public function configureOptions(OptionsResolver $resolver)
      * @throws \Exception
      */
-    public function buildViewContext(ViewContext $viewContext, DatalistFieldInterface $field, $row, array $options)
+    public function buildViewContext(ViewContext $viewContext, DatalistFieldInterface $field, mixed $value, array $options)
```
Missing return type

jmsche commented on src/Datalist/Field/Type/TextFieldType.php:
```diff
@@ -26,22 +20,16 @@ public function configureOptions(OptionsResolver $resolver)
-     * @param mixed $row
-     */
-    public function buildViewContext(ViewContext $viewContext, DatalistFieldInterface $field, $row, array $options)
+    public function buildViewContext(ViewContext $viewContext, DatalistFieldInterface $field, mixed $value, array $options)
```
Missing return type

jmsche commented on src/Datalist/Field/Type/UrlFieldType.php:
```diff
@@ -23,34 +21,25 @@ public function configureOptions(OptionsResolver $resolver)
-     * @param mixed $row
-     */
-    public function buildViewContext(ViewContext $viewContext, DatalistFieldInterface $field, $row, array $options)
+    public function buildViewContext(ViewContext $viewContext, DatalistFieldInterface $field, mixed $value, array $options)
```
Missing return type

jmsche commented on src/Datalist/Filter/DatalistFilter.php:
```diff
@@ -3,63 +3,37 @@
-     * @param mixed  $default
-     */
-    public function getOption($name, $default = null)
+    public function getOption(string $name, mixed $default = null)
```
Missing return type

jmsche commented on src/Datalist/Filter/DatalistFilterInterface.php:
```diff
@@ -2,43 +2,24 @@
-     * @param mixed  $default
-     */
-    public function getOption($name, $default = null);
+    public function getOption(string $name, mixed $default = null);
```
Missing return type

jmsche commented on src/Datalist/Filter/Expression/CombinedExpression.php:
```diff
@@ -7,27 +7,14 @@ class CombinedExpression implements ExpressionInterface
-     * @var string
-     */
-    private $operator;
+    private array $expressions;
```
Missing default value

bastien70 replied:
A default value is assigned in the constructor :

```php
    private array $expressions;

    public function __construct(private string $operator)
    {
        if (!\in_array($operator, self::getValidOperators(), true)) {
            throw new \InvalidArgumentException(sprintf('Unknown operator "%s"', $operator));
        }

        $this->expressions = \array_slice(\func_get_args(), 1);
    }
```

jmsche commented on src/Datalist/Filter/Type/AbstractFilterType.php:
```diff
@@ -18,10 +15,7 @@ public function configureOptions(OptionsResolver $resolver)
-     * @param mixed $row
-     */
-    public function buildViewContext(ViewContext $viewContext, DatalistFilterInterface $filter, $row, array $options)
+    public function buildViewContext(ViewContext $viewContext, DatalistFilterInterface $filter, mixed $row, array $options)
```
Missing return type

jmsche commented on src/Datalist/Filter/Type/ChoiceFilterType.php:
```diff
@@ -40,34 +37,22 @@ public function buildForm(FormBuilderInterface $builder, DatalistFilterInterface
-     * @param mixed $value
-     */
-    public function buildExpression(DatalistFilterExpressionBuilder $builder, DatalistFilterInterface $filter, $value, array $options)
+    public function buildExpression(DatalistFilterExpressionBuilder $builder, DatalistFilterInterface $filter, mixed $value, array $options)
```
Missing return type

jmsche commented on src/Datalist/Filter/Type/EntityFilterType.php:
```diff
@@ -47,35 +44,23 @@ public function buildForm(FormBuilderInterface $builder, DatalistFilterInterface
-     * @param mixed $value
-     */
-    public function buildExpression(DatalistFilterExpressionBuilder $builder, DatalistFilterInterface $filter, $value, array $options)
+    public function buildExpression(DatalistFilterExpressionBuilder $builder, DatalistFilterInterface $filter, mixed $value, array $options)
```
Missing return type

jmsche commented on src/Datalist/Filter/Type/SearchFilterType.php:
```diff
@@ -31,10 +28,7 @@ public function buildForm(FormBuilderInterface $builder, DatalistFilterInterface
-     * @param mixed $value
-     */
-    public function buildExpression(DatalistFilterExpressionBuilder $builder, DatalistFilterInterface $filter, $value, array $options)
+    public function buildExpression(DatalistFilterExpressionBuilder $builder, DatalistFilterInterface $filter, mixed $value, array $options)
```
Missing return type

jmsche commented on src/Datalist/TypeInterface.php:
```diff
@@ -9,15 +9,9 @@
-    public function getName();
+    public function getName(): string;
 
     public function configureOptions(OptionsResolver $resolver);
```
Missing return type

jmsche commented on src/Datalist/ViewContext.php:
```diff
@@ -14,20 +11,16 @@ class ViewContext implements \ArrayAccess
-     * @return bool
      */
-    public function offsetExists($offset)
+    public function offsetExists($offset): bool
```
Missing param type

jmsche commented on src/Datalist/ViewContext.php:
```diff
@@ -14,20 +11,16 @@ class ViewContext implements \ArrayAccess
-     * @return mixed
      */
-    public function offsetGet($offset)
+    public function offsetGet($offset): mixed
```
Missing param type

jmsche commented on src/Form/Type/FileType.php:
```diff
@@ -101,10 +92,7 @@ public function buildView(FormView $view, FormInterface $form, array $options)
-     * @param string $uploadDir
-     */
-    public function setUploadDir($uploadDir)
+    public function setUploadDir(string $uploadDir)
```
Missing return type

jmsche commented on src/Form/Type/ImageType.php:
```diff
@@ -4,25 +4,19 @@
-     * @return string|null
-     */
-    public function getParent()
+    public function getParent(): ?string
```
Can be "string"

jmsche commented on src/Form/Type/RecaptchaType.php:
```diff
@@ -12,20 +12,14 @@ class RecaptchaType extends AbstractRecaptchaType
-     * @var bool
      */
-    protected $ajax;
+    protected bool $ajax;
```
Can be initialized in constructor

jmsche commented on src/Form/Type/RecaptchaType.php:
```diff
@@ -58,15 +52,15 @@ public function configureOptions(OptionsResolver $resolver)
      * {@inheritdoc}
      */
-    public function getParent()
+    public function getParent(): ?string
```
Can be "string"

jmsche commented on src/Navigation/NavigationRegistry.php:
```diff
@@ -39,64 +24,44 @@ public function setActivePaths(array $paths)
-     * @param string $path
-     */
-    public function addActivePath($path)
+    public function addActivePath(string $path)
```
Missing return type

jmsche commented on src/Navigation/NavigationRegistry.php:
```diff
@@ -39,64 +24,44 @@ public function setActivePaths(array $paths)
-     * @param string $label
-     */
-    public function appendBreadcrumb($path, $label)
+    public function appendBreadcrumb(string $path, string $label)
```
Missing return type

jmsche commented on src/Navigation/NavigationRegistry.php:
```diff
@@ -39,64 +24,44 @@ public function setActivePaths(array $paths)
-     * @param string $label
-     */
-    public function prependBreadcrumb($path, $label)
+    public function prependBreadcrumb(string $path, string $label)
```
Missing return type

jmsche commented on src/Sitemap/AbstractSitemap.php:
```diff
@@ -95,11 +72,9 @@ protected function addUrl($loc, \DateTime $lastMod = null, $changeFreq = null, $
      * @throws \InvalidArgumentException
      */
-    private function validateChangeFreq($changeFreq)
+    private function validateChangeFreq(string $changeFreq)
```
Missing return type

jmsche commented on src/Sitemap/AbstractSitemap.php:
```diff
@@ -116,11 +91,9 @@ private function validateChangeFreq($changeFreq)
      * @throws \InvalidArgumentException
      */
-    private function validatePriority($priority)
+    private function validatePriority(string $priority)
```
Missing return type

jmsche commented on src/Twig/Extension/GoogleExtension.php:
```diff
@@ -63,55 +37,40 @@ public function getFunctions()
      * @param string $domainName Available options are "auto" or "none" or a real domain name
      */
-    public function setDomainName($domainName)
+    public function setDomainName(string $domainName)
```
Missing return type

jmsche commented on src/Twig/Extension/GoogleExtension.php:
```diff
@@ -63,55 +37,40 @@ public function getFunctions()
-     * @param string $allowLinker
-     */
-    public function setAllowLinker($allowLinker)
+    public function setAllowLinker(string $allowLinker)
```
Missing return type

jmsche commented on src/Twig/Extension/GoogleExtension.php:
```diff
@@ -63,55 +37,40 @@ public function getFunctions()
-     * @param string $tagsManagerId
-     */
-    public function setTagsManagerId($tagsManagerId)
+    public function setTagsManagerId(string $tagsManagerId)
```
Missing return type

jmsche commented on src/Twig/Extension/NavigationExtension.php:
```diff
@@ -63,48 +52,31 @@ public function addActivePath($path)
-     * @param string $label
-     */
-    public function appendBreadcrumb($path, $label)
+    public function appendBreadcrumb(string $path, string $label)
```
Missing return type

jmsche commented on src/Twig/Extension/NavigationExtension.php:
```diff
@@ -63,48 +52,31 @@ public function addActivePath($path)
-     * @param string $label
-     */
-    public function prependBreadcrumb($path, $label)
+    public function prependBreadcrumb(string $path, string $label)
```
Missing return type

jmsche commented on src/Twig/Extension/PaginatorExtension.php:
```diff
@@ -81,10 +52,7 @@ public function renderPaginatorWidget(Environment $env, PaginatorInterface $pagi
-     * @param string $template
-     */
-    public function setTemplate($template)
+    public function setTemplate(string $template)
```
Missing return type

jmsche commented on src/Twig/Extension/SiteExtension.php:
```diff
@@ -71,28 +47,17 @@ public function getPageTitle($baseTitle, $seperator = ' - ')
-     * @param string $description
-     */
-    public function setMetaDescription($description)
+    public function setMetaDescription(string $description)
```
Missing return type

jmsche commented on src/Twig/Extension/SiteExtension.php:
```diff
@@ -109,18 +74,12 @@ public function addMetaKeywords(array $keywords)
-     * @param string $prepend
-     */
-    public function prependPageTitle($prepend)
+    public function prependPageTitle(string $prepend)
```
Missing return type

jmsche commented on src/Twig/Extension/SiteExtension.php:
```diff
@@ -109,18 +74,12 @@ public function addMetaKeywords(array $keywords)
-     * @param string $append
-     */
-    public function appendPageTitle($append)
+    public function appendPageTitle(string $append)
```
Missing return type

jmsche commented on src/Twig/Extension/TextExtension.php:
```diff
@@ -171,10 +154,8 @@ function ($matches) {
-     * @param bool $useMultiByteString
      */
-    public function setMultiByteString($useMultiByteString)
+    public function setMultiByteString(bool $useMultiByteString)
```
Missing return type

bastien70 committed:
Fixed review

jmsche commented on src/Datalist/Field/Type/FieldTypeInterface.php:
```diff
@@ -2,17 +2,13 @@
-     * @param mixed $value
-     */
-    public function buildViewContext(ViewContext $viewContext, DatalistFieldInterface $field, $value, array $options);
+    public function configureOptions(OptionsResolver $resolver): void;
```
I don't understand this change

bastien70 replied:
Yes, I had made a global find and replace and had made a mistake for this function. I must have forgotten to rollback this file, thanks for finding it

jmsche commented on src/Datalist/Filter/DatalistFilterInterface.php:
```diff
@@ -2,43 +2,26 @@
-     * @param mixed  $default
-     */
-    public function getOption($name, $default = null);
+    public function getOption(string $name, mixed $default = null): ?string;
```
Return type should be mixed I think

jmsche commented on src/Form/Type/RecaptchaType.php:
```diff
@@ -10,28 +10,15 @@
-    protected $localeResolver;
-
-    public function __construct($publicKey, $enabled, $ajax, LocaleResolver $localeResolver, $apiHost = 'www.google.com')
+    public function __construct($publicKey, $enabled, protected $ajax, protected LocaleResolver $localeResolver, $apiHost = 'www.google.com')
```
protected bool $ajax

bastien70 committed:
Fixed review

jmsche commented:
Thank you @bastien70.
